src/astro/databases/snowflake.py

```diff
@@ -57,6 +76,252 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
+            if get_filetype(
+                source_file.path
+            ) == FileType.PARQUET and not self.table_exists(table=target_table):
+                self.create_table_from_parquet(
+                    source_file=source_file,
+                    target_table=target_table,
+                    snowflake_hook=snowflake_hook,
+                )
+            else:
+                return False
+
+        # check if snowflake table exists
+        if self.table_exists(table=target_table):
+            logging.info("Table exists.")
+            stage_name = self.create_stage(
+                source_file=source_file, snowflake_hook=snowflake_hook
+            )
+            sql_parts = self.create_copy_into_sql_parts(
+                source_file=source_file, target_table=target_table, stage=stage_name
+            )
+            copy_query = "\n".join(sql_parts)
+
+            logging.info("Executing COPY command...")
+            execution_info = snowflake_hook.run(
+                copy_query, autocommit=kwargs.get("autocommit", True)
+            )
+            logging.info("COPY command completed")
+            logging.info(execution_info)
+            self.drop_stage(stage_name=stage_name, snowflake_hook=snowflake_hook)
+            return True
+        return False
+
+    def create_stage(self, source_file: File, snowflake_hook: SnowflakeHook) -> str:
+        """
+        Creates a stage name based on source_file path and storage integration provided.
+
+        @param source_file: Source file
+        @param snowflake_hook: Snowflake hook
+        @return: Stage name created
+        """
+        # Here,
+        # To copy dataset from gcs_path = gs://sample-bucket/workspace/sample.csv" to snowflake table
+        # where stage is WORKSPACE_STAGE, following steps has to be completed.
+
+        # The path of stage should be to the folder the file is in. Command to create STAGE would look
+        # like following:
+        # CREATE OR REPLACE STAGE <warehouse_name>.WORKSPACE_STAGE URL='gcs://sample-bucket/workspace'
+        # FILE_FORMAT=(TYPE=CSV, TRIM_SPACE=TRUE) COPY_OPTIONS=(MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE)
+        # storage_integration = <storage_integration_name>;
+
+        # COPY INTO query would be:
+        # COPY INTO sample_table from '@WORKSPACE_STAGE/sample.csv';
+        FILE_FORMAT_MAPPING = {
+            FileType.CSV: "CSV",
+            FileType.PARQUET: "PARQUET",
+            FileType.NDJSON: "JSON",
+        }
+        snowflake_query_file_storage_mapping = {"gs": "gcs", "s3": "s3"}
+        stage_name = (
+            "stage_"
+            + random.choice(string.ascii_lowercase)
+            + "".join(
+                random.choice(string.ascii_lowercase + string.digits) for _ in range(7)
+            )
+        )
+
+        parsed_url = urlparse(source_file.path, allow_fragments=False)
+        bucket_type = snowflake_query_file_storage_mapping.get(
+            parsed_url.scheme.lower(), "gcs"
```

Comment:
Do we want to default the bucket_type to `gcs`? Perhaps we should raise an exception if it is not one of the pre-defined supported ones.

Reply:
[Done]([Add exception handling and change return types](https://github.com/astronomer/astro-sdk/pull/487/commits/863c4e9c8160f91412565a52aff93939f680638d))